streaming/base/dataset.py

```diff
@@ -189,37 +196,51 @@ def __init__(self,
         # Determine and distribute shuffle seed and shm prefix.
         seed_rng = np.random.default_rng(shuffle_seed)
         self.shuffle_seed = int(seed_rng.integers(1 << 60))
-        prefix_int = np.random.randint(1 << 24)
+        prefix_int = int(seed_rng.integers(1 << 24))
         self._prefix = f'{prefix_int:06x}'
 
+        # Create the shared memory-backed worker barrier, without its lock, which is unpickleable.
+        worker_barrier_filelock_path = os.path.join(os.path.sep, 'tmp', 'streaming', self._prefix,
+                                                    'barrier_filelock')
+        worker_barrier_shm_path = f'{self._prefix}_barrier'
+
+        # Should be a unique shared memory block filename per each StreamingDataset instantiation to avoid a conflict
+        # between a different StreamingDataset instance on a same machine.
+        if self.world.is_local_leader:
+            if os.path.exists(os.path.dirname(worker_barrier_filelock_path)):
+                raise ValueError(''.join([
```

Comment:
Is there a way to avoid this? It's not great if instantiating two datasets (eg train and val) with default Paramus raises an error. Maybe we keep the prefix as unseeded randomness?

Reply:
Ah yeah, can we just pick the random prefix in a loop until available

Reply:
Yes, I think we can iterate until we get a unique `prefix` and avoid raising an exception. The reason we need the unique `prefix` and had that exception was that it was throwing some nasty exception where the exception traceback was altogether showing at a different place which makes it harder to root-cause. However, having a unique `prefix` does not have this issue. 

Performing the checking and re-generating the unique and unused prefix is a bit challenging without using a distributed barrier and can cause some bug. Hence, I will add this feature in the upcoming PR. For now, we will avoid raising an exception.